### DISCUSSION TITLE
Check data size before saving the entry to avoid MySQL error

### DISCUSSION THREAD

**Discussion by PaulBaileyAtChan** [ExpressionEngine/ExpressionEngine#3381] [Feature Requests]
The issue of EE hitting a MySQL exception on saving field data larger than 64KB (#464) was addressed in #467 (EE5.4.0), by adding the option to use a MEDIUMTEXT (16MB) column size. This helped, but was only a partial solution.

Saving 64KB+ field data, in a field that was created using the TEXT (64KB) column size, still results in a MySQL exception in EE6.4.3. Discussion of the proposed solution in EE5.4.0 said that:

> In EE 6 we're also adding an additional JS check on the publish / edit screen for content length on appropriate native fields and will warn the user accordingly.

It doesn't look like this has been implemented yet, so there's nothing in EE to prevent a MySQL exception.

Minimally, as discussed, EE should anticipate this issue prior to saving, and fail gracefully, providing the user with appropriate guidance to deal with the situation. Providing the MEDIUMTEXT option alone is a rough patch, rather than a complete solution. A user might not anticipate when a field is created how large the data might become, and asking a typical user to understand how to increase the column size isn't reasonable or user-friendly.

(_Ideally_, EE should manage this implementation detail invisibly, increasing the column size automatically, or splitting the data across multiple fields, or whatever solution makes most sense.)

**Comment by intoeetive** [comment]
Giving it more generic name and converting to feature request
